server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -265,6 +266,23 @@ public CopyTo copyTo() {
         return copyTo;
     }
 
+    private String readComplexJsonElement(XContentParser parser, int tokenDepth) throws IOException {
+        XContentParser.Token closingToken = null;
+        if (parser.currentToken() == XContentParser.Token.START_ARRAY) {
+            closingToken = XContentParser.Token.END_ARRAY;
+        } else if (parser.currentToken() == XContentParser.Token.START_OBJECT) {
+            closingToken = XContentParser.Token.END_OBJECT;
+        } else throw new IOException("Incorrect token type");
+
+        StringBuilder builder = new StringBuilder();
+        for (int i = 0;i < tokenDepth
+            && parser.currentToken() != null
+            && parser.currentToken() != XContentParser.Token.END_OBJECT;i++) {
```

Comment:
did you intend to use `closingToken` here instead of  `XContentParser.Token.END_OBJECT`?

Reply:
oops